device/cuda/src/cca/component_connection.cu

```diff
@@ -764,16 +763,49 @@ component_connection::output_type component_connection::operator()(
         }
     }
 
+    /*
+     * Allocate space on the device for input.
+     */
+    vecmem::unique_alloc_ptr<channel_id[]> channel0_device =
+        vecmem::make_unique_alloc<channel_id[]>(dmr, channel0.size());
+    vecmem::unique_alloc_ptr<channel_id[]> channel1_device =
+        vecmem::make_unique_alloc<channel_id[]>(dmr, channel1.size());
+    vecmem::unique_alloc_ptr<scalar[]> activation_device =
+        vecmem::make_unique_alloc<scalar[]>(dmr, activation.size());
+    vecmem::unique_alloc_ptr<scalar[]> time_device =
+        vecmem::make_unique_alloc<scalar[]>(dmr, time.size());
+    vecmem::unique_alloc_ptr<geometry_id[]> module_id_device =
+        vecmem::make_unique_alloc<geometry_id[]>(dmr, module_id.size());
+
+    /*
+     * Move data from host to device.
+     */
+    CUDA_ERROR_CHECK(cudaMemcpy(channel0_device.get(), channel0.data(),
+                                total_cells * sizeof(channel_id),
+                                cudaMemcpyHostToDevice));
+    CUDA_ERROR_CHECK(cudaMemcpy(channel1_device.get(), channel1.data(),
+                                total_cells * sizeof(channel_id),
+                                cudaMemcpyHostToDevice));
+    CUDA_ERROR_CHECK(cudaMemcpy(activation_device.get(), activation.data(),
+                                total_cells * sizeof(scalar),
+                                cudaMemcpyHostToDevice));
+    CUDA_ERROR_CHECK(cudaMemcpy(time_device.get(), time.data(),
+                                total_cells * sizeof(scalar),
+                                cudaMemcpyHostToDevice));
+    CUDA_ERROR_CHECK(cudaMemcpy(module_id_device.get(), module_id.data(),
+                                total_cells * sizeof(geometry_id),
+                                cudaMemcpyHostToDevice));
```

Comment:
Of course if using vector buffers, then these copies could be made with `vecmem::cuda::copy`.